cfme/physical/physical_server.py

```diff
@@ -154,6 +160,66 @@ def wait_for_delete(self):
             fail_func=view.browser.refresh
         )
 
+    def validate_stats(self, ui=False):
+        """ Validates that the detail page matches the physical server's information.
+
+        This method logs into the provider using the mgmt_system interface and collects
+        a set of statistics to be matched against the UI. An exception will be raised
+        if the stats retrieved from the UI do not match those retrieved from wrapanapi.
+        """
+
+        # Make sure we are on the physical server detail page
+        if ui:
+            self.load_details()
+
+        # Retrieve the data from the provider
+        providers_data = conf.cfme_data.get("management_systems", {})
```

Comment:
**Required**

I think this should be pulled out into separate properties for `data` and `mgmt`, instead of burying them here in validate_stats. Otherwise we'll end up with additional methods that also need the yaml data, or need a mgmt class object.

See how it's handled in the BaseProvider object:
https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/provider.py#L68

I'd like to see something similar here for this object. One immediate benefit could be better handling of error conditions - these lines could all generate KeyError/AttributeError.

Reply:
Taking a 2nd look, this actually looks like you'll get the exact same object as if you used `self.provider.mgmt` since you're fetching the provider's yaml data.  If this is a separate record in the yaml, then you'll need specific methods on this class.

In general - the `'lenovo'` hardcoded string should not be used here, and this should be a class attribute. This could be _any_ 'lenovo' key in the management_systems